backend/src/LibExecution/ProgramTypes.fs

```diff
@@ -396,19 +394,19 @@ type Expr =
   | EInfix of id * Infix * Expr * Expr
   // the id in the varname list is the analysis id, used to get a livevalue
   // from the analysis engine
-  | ELambda of id * List<id * string> * Expr
+  | ELambda of id * NEList<id * string> * Expr
   | EFieldAccess of id * Expr * string
   | EVariable of id * string
   | EApply of id * Expr * typeArgs : List<TypeReference> * args : List<Expr>
   | EFnName of id * NameResolution<FnName.T>
   | EList of id * List<Expr>
   | EDict of id * List<string * Expr>
   | ETuple of id * Expr * Expr * List<Expr>
-  | EPipe of id * Expr * PipeExpr * List<PipeExpr>
+  | EPipe of id * Expr * NEList<PipeExpr>
```

Comment:
NEList seems fine here, as we already had one in practice.

One place this wouldn't support is:
```fsharp
5 |> // user omitted the code to the right
```

this code won't parse. If `|>` was a simple infix fn application, than the code would parse. As such, I think going back to a List here might be appropriate.